dags/inference/configs/trt_llm_mlperf_v40_config.py

```diff
@@ -0,0 +1,153 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities to construct configs for MLPerf4.0 Reproduce DAG."""
+
+import datetime
+from typing import Dict
+from xlml.apis import gcp_config, metric_config, task, test_config
+from dags import test_owner, vm_resource
+from dags.vm_resource import Project, RuntimeVersion
+
+RUNTIME_IMAGE = RuntimeVersion.TPU_UBUNTU2204_BASE.value
+GCS_SUBFOLDER_PREFIX = test_owner.Team.INFERENCE.value
+
+
+def get_trt_llm_mlperf_v40_gpu_config(
+    machine_type: vm_resource.MachineVersion,
+    image_project: vm_resource.ImageProject,
+    image_family: vm_resource.ImageFamily,
+    accelerator_type: vm_resource.GpuVersion,
+    count: int,
+    gpu_zone: vm_resource.Zone,
+    time_out_in_min: int,
+    test_name: str,
+    project: Project,
+    network: str,
+    subnetwork: str,
+    model_configs: Dict = {},
+) -> task.GpuCreateResourceTask:
+  docker_container_name = "mlperf-inference"
+  set_up_cmds = (
+      # Install Nvidia driver
+      "wget -c https://us.download.nvidia.com/tesla/550.54.15/NVIDIA-Linux-x86_64-550.54.15.run",
+      "chmod u+x NVIDIA-Linux-x86_64-550.54.15.run",
+      "sudo ./NVIDIA-Linux-x86_64-550.54.15.run -x-module-path=/usr/lib/xorg/modules --ui=none -x-library-path=/usr/lib -q",
+      "sudo nvidia-smi -pm 1",
+      # Format and mount multiple Local SSD
+      "sudo apt update && sudo apt install mdadm --no-install-recommends",
+      "find /dev/ | grep google-local-nvme-ssd",
+      "sudo mdadm --create /dev/md0 --level=0 --raid-devices=16 /dev/disk/by-id/google-local-nvme-ssd-0 /dev/disk/by-id/google-local-nvme-ssd-1 /dev/disk/by-id/google-local-nvme-ssd-2 /dev/disk/by-id/google-local-nvme-ssd-3 /dev/disk/by-id/google-local-nvme-ssd-4 /dev/disk/by-id/google-local-nvme-ssd-5 /dev/disk/by-id/google-local-nvme-ssd-6 /dev/disk/by-id/google-local-nvme-ssd-7 /dev/disk/by-id/google-local-nvme-ssd-8 /dev/disk/by-id/google-local-nvme-ssd-9 /dev/disk/by-id/google-local-nvme-ssd-10 /dev/disk/by-id/google-local-nvme-ssd-11 /dev/disk/by-id/google-local-nvme-ssd-12 /dev/disk/by-id/google-local-nvme-ssd-13 /dev/disk/by-id/google-local-nvme-ssd-14 /dev/disk/by-id/google-local-nvme-ssd-15",
+      "sudo mdadm --detail --prefer=by-id /dev/md0",
+      "sudo mkfs.ext4 -F /dev/md0",
+      "sudo mkdir -p /scratch",
+      "sudo mount /dev/md0 /scratch",
+      "sudo chmod a+w /scratch",
+      "cd /scratch",
+      # Prepare data
+      "mkdir -p models",
+      "mkdir -p preprocessed_data",
+      "gsutil -m cp -n -r gs://tohaowu/mlpinf-v40/models/Llama2 models/",
+      "gsutil -m cp -n -r gs://tohaowu/mlpinf-v40/preprocessed_data/open_orca preprocessed_data/",
```

Comment:
Can you please explain how this data was generated? Was it generated for last round?

I think the data (and models) should be the same for this upcoming round. So should we rename this, or make a copy of the data into a new GCS bucket to be more clear?

Reply:
@tohaowu I think this data is MLPerf4.0's round, is this same as MLPerf4.1's ? Thanks!